server/routerlicious/packages/local-server/src/localWebSocketServer.ts

```diff
@@ -37,6 +37,11 @@ export class LocalWebSocket implements core.IWebSocket {
     }
 
     public emit(event: string, ...args: any[]) {
+        // Disconnect from the "socket" if the message is greater than 1MB
+        if (JSON.stringify(args).length > 1e6) {
+            this.disconnect();
```

Comment:
might want to add a `return;` after this so the emit line won't get called?

Reply:
good call, for some reason I had decided the return wasn't necessary. looks fairly obvious to me now that it is 😄

Reply:
@GaryWilber, @justus-camp - this change causes trouble. What I see is that local tests (running in-memory server) fail for large payloads as the "server" accumulates a ton of ops (each of them is smaller than 1Mb) and hits that check, getting into infinite loop of disconnects & reconnects (maybe it's not infinite, but test stops making any progress).
And if I comment this code out, everything works fine.

Repro is simple - a UT that sends a ton of largish ops (anything in 250K - 800K range). Batching is done in runtime (not using protocol capability), so every op shows up as one op in a batch.

@NicholasCouri - FYI

Reply:
I have a feeling that what you wanted is on inbound direction (client side) to put that limit, not on outbound (service side). But I think same class is used for both for in-memory server, and it screws everything.